pytensor/tensor/rewriting/linalg.py

```diff
@@ -310,3 +311,27 @@ def local_log_prod_sqr(fgraph, node):
 
         # TODO: have a reduction like prod and sum that simply
         # returns the sign of the prod multiplication.
+
+
+cholesky_no_inplace = Cholesky(overwrite_a=False)
+cholesky_inplace = Cholesky(overwrite_a=True)
+
+
+@node_rewriter([cholesky_no_inplace], inplace=True)
```

Comment:
Cholesky has more flags like the nan policy, so this won't work. Instead:

```suggestion
@node_rewriter([Cholesky], inplace=True)
```

And inside the rewrite check it's not already destructive and recreate the Op with the same flags for nan but `destructive=True`. 

We can probably use the same logic that the random_variable_inplace uses as that also only changes one flag out of many. Actually we could refactor that into a function that does that for any op whose destructive flag is just a boolean and needs no other logic. 

```python
def make_destructive(node, destructive_prop="destructive"):
  op = node.op
  props = op.props()  # or get_props? never remember 
  if props[destructive_prop]:
    return False  # already destructive
  props[destructive_prop] = True
  destructive_op = type(op)(**props)
  return destructive_op.make_node(node.inputs). outputs
```

Then your rewrite would simply be

```python
@node_rewriter([Cholesky], inplace=True)
def destructive_cholesky(fgraph, node):
  return make_destructive(node)
```

And exactly the same for the RV one and many (most) others that need no extra logic?

Reply:
It was `op._prop_dict()` -- as horrible a name as possible.

I can refactor all the other destructive rewrites to use `make_inplace` in this PR if you want, or open a new one if that's a bit more organized.

Reply:
A separate PR for the others sounds good